src/HRQLS/Models/ElasticSearchServiceProvider.php

```diff
@@ -82,12 +84,188 @@ public function boot(Application $app)
+     * @param array $indices The list of indices to include in the query. Defaults to searching all indices.
+     * @param array $types   The list of types to include in the query. Defaults to searching all types.
+     * @param array $query   The query to run against the specified indexes and types.
      *
-     * @return array
+     * @return array Like [
+     *    'took' => The amount of time the query took to execute,
+     *    'timed_out' => [
+     *      '_shards' => [
+     *        'total' => (Integer), Total number of shards queried,
+     *        'successful' => (integer), Number of successfully queried shards.
+     *        'failed' => (integer), Number of shards that could not be queried.
+     *    ],
+     *    'hits' => [
+     *      'total' => (integer), Number of documents returned by the query.
+     *      'max_score' => (double), The highest score assigned during execution.
+     *      'hits' => [
+     *      ],
+     *    ],
+     *  ];
      */
-    public function search(array $req)
+    public function search(array $indices, array $types, array $query = [])
     {
+        if (empty($indices)) {
+            $indices = ['_all'];
+        }
+        
+        if ($types === null) {
+            $types = [];
+        }
+        
+        $req = [
+            'index' => implode(',', $indices),
+            'types' => implode(',', $types),
+            'body' => $query,
+        ];
+        
         return $this->client->search($req);
     }
+    
+    /**
+     * Inserts (a.k.a indexes) the document into the specified index.
+     *
+     * @param string  $index The name of the index to insert the document.
+     * @param string  $type  The type (a.k.a collection) to insert the document.
+     * @param array   $doc   The document to insert.
+     * @param integer $id    The Id to assign the document being inserted. If null Id is auto-assigned (recommended).
+     *
+     * @return array Like [
+     *    '_shards' => [
+     *        'total' => (Integer),
+     *        'failed' => (Integer),
+     *        'successful' = > (Integer),
+     *    ],
+     *    '_index' => (String),
+     *    '_type' => (String),
+     *    '_id' => (String),
+     *    '_version' => (Integer),
+     *    'created' => (Boolean),
+     *  ];
+     *
+     * @throws UsageException When $index, $type, or $doc is null.
+     */
+    public function insert($index, $type, array $doc, $id = null)
+    {
+        $errorMessages = [];
+        
+        if (empty($index)) {
+            $errorMessages[] = '$index cannot be null.';
+        }
+        
+        if (empty($type)) {
+            $errorMessages[] = '$type cannot be null.';
+        }
+
+        if (empty($doc)) {
+            $errorMessages[] = "An empty document cannot be inserted. That doesn't make any sense.";
+        }
+
+        if (!empty($errorMessages)) {
+            throw new UsageException(implode("\n", $errorMessages));
+        }
+
+        $req = [
```

Comment:
Do a git blame and you'll see you are the one that started that naming convention in the Service Provider.